### ISSUE TITLE
Unable to Reproduce the Results

### ISSUE THREAD

**Issue by kevintsq** [eric-zqwang/puzzlefusion-plusplus#5]
I tried to reproduce the results on a single GPU but they were not satisfactory. The metrics are as follows:
```
┏━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
┃        Test metric        ┃       DataLoader 0        ┃
┡━━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━━┩
│       eval/part_acc       │    0.36521467566490173    │
│        eval/rmse_r        │    59.105369567871094     │
│        eval/rmse_t        │    0.20479142665863037    │
│       eval/shape_cd       │    0.06885631382465363    │
└───────────────────────────┴───────────────────────────┘
```
A sample video is as follows:

https://github.com/user-attachments/assets/d7989074-47a8-47ce-a815-f4e694c9a9cf

I'm wondering why. I didn't modify anything except for the training batch size of vqvae.

**Comment by eric-zqwang** [comment]
Hi,

Thanks for interest. Do you try use the pre-trained checkpoints? Can you reproduce results using the pre-trained checkpoints?

> I'm wondering why. I didn't modify anything except for the training batch size of vqvae.

Can you tell me more details? Do you train the 3 modules from scratch (i.e. vqvae, denoiser, verifier)? Or you just re-trained the VQVAE then can not reproduce results.

**Comment by kevintsq** [comment]
Thanks for your prompt reply. The reproduction from the pre-trained checkpoints was successful. However, the results are not good if I train the 3 modules from scratch.

**Comment by eric-zqwang** [comment]
Ah I see. Can you show me the wandb log for training the denoiser? The denoiser also log the metric for denoiser only results.

**Comment by kevintsq** [comment]
The `debug.log` is as follows:
```
2024-09-10 20:56:06,837 INFO    MainThread:139288 [wandb_setup.py:_flush():77] Current SDK version is 0.17.9
2024-09-10 20:56:06,837 INFO    MainThread:139288 [wandb_setup.py:_flush():77] Configure stats pid to 139288
2024-09-10 20:56:06,837 INFO    MainThread:139288 [wandb_setup.py:_flush():77] Loading settings from /home/administrator/.config/wandb/settings
2024-09-10 20:56:06,837 INFO    MainThread:139288 [wandb_setup.py:_flush():77] Loading settings from /home/administrator/PycharmProjects/puzzlefusion-plusplus/wandb/settings
2024-09-10 20:56:06,837 INFO    MainThread:139288 [wandb_setup.py:_flush():77] Loading settings from environment variables: {}
2024-09-10 20:56:06,837 INFO    MainThread:139288 [wandb_setup.py:_flush():77] Applying setup settings: {'_disable_service': False}
2024-09-10 20:56:06,837 INFO    MainThread:139288 [wandb_setup.py:_flush():77] Inferring run settings from compute environment: {'program_relpath': 'train_denoiser.py', 'program_abspath': '/home/administrator/PycharmProjects/puzzlefusion-plusplus/train_denoiser.py', 'program': '/home/administrator/PycharmProjects/puzzlefusion-plusplus/train_denoiser.py'}
2024-09-10 20:56:06,837 INFO    MainThread:139288 [wandb_setup.py:_flush():77] Applying login settings: {}
2024-09-10 20:56:06,837 INFO    MainThread:139288 [wandb_setup.py:_flush():77] Applying login settings: {'mode': 'offline'}
2024-09-10 20:56:06,837 INFO    MainThread:139288 [wandb_init.py:_log_setup():524] Logging user logs to /home/administrator/PycharmProjects/puzzlefusion-plusplus/output/denoiser/everyday_epoch2000_bs64/training/wandb/offline-run-20240910_205606-macema9o/logs/debug.log
2024-09-10 20:56:06,837 INFO    MainThread:139288 [wandb_init.py:_log_setup():525] Logging internal logs to /home/administrator/PycharmProjects/puzzlefusion-plusplus/output/denoiser/everyday_epoch2000_bs64/training/wandb/offline-run-20240910_205606-macema9o/logs/debug-internal.log
2024-09-10 20:56:06,837 INFO    MainThread:139288 [wandb_init.py:init():608] calling init triggers
2024-09-10 20:56:06,837 INFO    MainThread:139288 [wandb_init.py:init():615] wandb.init called with sweep_config: {}
config: {}
2024-09-10 20:56:06,837 INFO    MainThread:139288 [wandb_init.py:init():658] starting backend
2024-09-10 20:56:06,837 INFO    MainThread:139288 [wandb_init.py:init():662] setting up manager
2024-09-10 20:56:06,838 INFO    MainThread:139288 [backend.py:_multiprocessing_setup():105] multiprocessing start_methods=fork,spawn,forkserver, using: spawn
2024-09-10 20:56:06,838 INFO    MainThread:139288 [wandb_init.py:init():670] backend started and connected
2024-09-10 20:56:06,839 INFO    MainThread:139288 [wandb_init.py:init():768] updated telemetry
2024-09-10 20:56:06,841 INFO    MainThread:139288 [wandb_init.py:init():801] communicating run to backend with 90.0 second timeout
2024-09-10 20:56:06,844 INFO    MainThread:139288 [wandb_init.py:init():852] starting run threads in backend
2024-09-10 20:56:08,814 INFO    MainThread:139288 [wandb_run.py:_console_start():2465] atexit reg
2024-09-10 20:56:08,814 INFO    MainThread:139288 [wandb_run.py:_redirect():2311] redirect: wrap_raw
2024-09-10 20:56:08,814 INFO    MainThread:139288 [wandb_run.py:_redirect():2376] Wrapping output streams.
2024-09-10 20:56:08,814 INFO    MainThread:139288 [wandb_run.py:_redirect():2401] Redirects installed.
2024-09-10 20:56:08,814 INFO    MainThread:139288 [wandb_init.py:init():895] run started, returning control to user process
2024-09-10 20:56:08,929 INFO    MainThread:139288 [wandb_run.py:_config_callback():1392] config_cb None None {'cfg': "{'project_root_path': '${hydra:runtime.cwd}', 'experiment_output_path': '${project_root_path}/output/denoiser/${experiment_name}', 'ckpt_path': None, 'experiment_name': 'everyday_epoch2000_bs64', 'train_seed': 123, 'test_seed': 123, 'logger': {'_target_': 'lightning.pytorch.loggers.WandbLogger', 'project': 'puzzlefusion_plusplus', 'name': '${experiment_name}', 'save_dir': '${experiment_output_path}/training'}, 'trainer': {'accelerator': 'gpu', 'max_epochs': 2000, 'num_sanity_val_steps': 1, 'check_val_every_n_epoch': 100, 'profiler': 'simple', 'precision': 32}, 'checkpoint_monitor': {'_target_': 'lightning.pytorch.callbacks.ModelCheckpoint', 'monitor': 'eval/part_acc', 'mode': 'max', 'save_last': True, 'save_top_k': 3, 'every_n_epochs': '${trainer.check_val_every_n_epoch}', 'filename': '{epoch}', 'dirpath': '${experiment_output_path}/training'}, 'ae': {'ae_name': {'_target_': 'puzzlefusion_plusplus.denoiser.model.modules.encoder.VQVAE'}, 'n_embeddings': 1024, 'embedding_dim': 16, 'num_point': 25, 'num_dim': 64, 'local_decode_pts': 40, 'beta': 0.25}, 'data': {'batch_size': 32, 'val_batch_size': 32, 'num_workers': 10, 'data_fn': 'everyday.{}.txt', 'data_dir': './data/pc_data/everyday/train/', 'data_val_dir': './data/pc_data/everyday/val/', 'rot_range': -1, 'overfit': -1, 'min_num_part': 2, 'max_num_part': 20}, 'model': {'model_name': {'_target_': 'puzzlefusion_plusplus.denoiser.model.denoiser.Denoiser'}, 'encoder_weights_path': 'output/autoencoder/everyday_2000epoch/training/last.ckpt', 'multiple_ref_parts': True, 'num_dim': 64, 'num_point': 25, 'out_channels': 7, 'std': 1, 'multires': 10, 'embed_dim': 512, 'num_layers': 6, 'num_heads': 8, 'dropout_rate': 0.1, 'DDPM_TRAIN_STEPS': 1000, 'DDPM_BETA_SCHEDULE': 'linear', 'timestep_spacing': 'leading', 'PREDICT_TYPE': 'epsilon', 'BETA_START': 0.0001, 'BETA_END': 0.02, 'num_inference_steps': 20, 'lr_scheduler': {'_target_': 'torch.optim.lr_scheduler.MultiStepLR', 'milestones': [1200, 1700], 'gamma': 0.5}}}"}
2024-09-10 21:41:33,139 WARNING MsgRouterThr:139288 [router.py:message_loop():77] message_loop has been closed
```
Actually I didn't use `wandb` to visualize the results. I'll enable it and train again to see whether it provides more information.

**Comment by eric-zqwang** [comment]
No worries. Can you send me a copy of your checkponts and codes via email? I'll test it on my side. My email is zwa170@sfu.ca

**Comment by kevintsq** [comment]
Hi, I shared the files with you via Google Drive. Thanks for your assistance!

**Comment by kevintsq** [comment]
It turns out that I used an incomplete dataset provided by others. Sorry for the inconvenience.
